From f1c3ed077f731ea219bf4ac182cc8e422f364eed Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Thu, 20 Aug 2015 18:43:29 -0700
Subject: [PATCH 015/206] SnapdragonCamera: Hide UI after error-checking video
 preferences

* Hiding the UI before error-checking the video preferences when
  video recording starts results in a broken UI when one of the
  error cases evaluates to be true.
* Hide the UI after the error checking to prevent the UI from
  breaking when an error is found.

Change-Id: Ia98798dbf2a33800d47497630714af17865767c6
---
 src/com/android/camera/VideoModule.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 1f9939c16..df66f36ed 100755
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1856,7 +1856,6 @@ public class VideoModule implements CameraModule,
         mStartRecPending = true;
         mUI.cancelAnimations();
         mUI.setSwipingEnabled(false);
-        mUI.hideUIwhileRecording();
         // When recording request is sent before starting preview, onPreviewFrame()
         // callback doesn't happen so removing preview cover here, instead.
         if (mUI.isPreviewCoverVisible()) {
@@ -1929,6 +1928,8 @@ public class VideoModule implements CameraModule,
             return false;
         }
 
+        mUI.hideUIwhileRecording();
+
         // Make sure the video recording has started before announcing
         // this in accessibility.
         AccessibilityUtils.makeAnnouncement(mUI.getShutterButton(),
-- 
2.17.1

